core/vm/environment.libevm.go

```diff
@@ -36,13 +37,26 @@ type environment struct {
 	callType CallType
 }
 
+func (e *environment) Gas() uint64            { return e.self.Gas }
+func (e *environment) UseGas(gas uint64) bool { return e.self.UseGas(gas) }
+func (e *environment) Value() *uint256.Int    { return new(uint256.Int).Set(e.self.Value()) }
+
 func (e *environment) ChainConfig() *params.ChainConfig  { return e.evm.chainConfig }
 func (e *environment) Rules() params.Rules               { return e.evm.chainRules }
 func (e *environment) ReadOnlyState() libevm.StateReader { return e.evm.StateDB }
 func (e *environment) IncomingCallType() CallType        { return e.callType }
 func (e *environment) BlockNumber() *big.Int             { return new(big.Int).Set(e.evm.Context.BlockNumber) }
 func (e *environment) BlockTime() uint64                 { return e.evm.Context.Time }
 
+func (e *environment) refundGas(add uint64) error {
+	gas, overflow := math.SafeAdd(e.self.Gas, add)
+	if overflow {
+		return ErrGasUintOverflow
```

Comment:
We should wrap errors to force callers to use `errors.Is` instead of `==` in case one layer adds a wrapping context with `fmt.Errorf` (now and in the future)

```suggestion
		return fmt.Errorf("%w", ErrGasUintOverflow) 
```

Reply:
While in theory I agree with you*, there is a lot of direct error checking in `geth` in general so I don't want to stray from that pattern.

*The raw `%w` without anything else is quite interesting to force `errors.Is`!

Reply:
> *The raw %w without anything else is quite interesting to force errors.Is!

Yeah it's talked about https://go.dev/blog/go1.13-errors#:~:text=If%20a%20function%20is%20defined%20as%20returning%20an%20error%20wrapping%20some%20sentinel%20or%20type%2C%20do%20not%20return%20the%20underlying%20error%20directly. (interesting read too!)